add workflow

namidapoo opened the pull request.

namidapoo commented:
@kenfdev テストを追加したところワークフローが失敗してしまいます
ローカルで実行したときも、失敗した後また実行したらパスしたりします
また、わざと落ちるテストを書いた時のエラーメッセージもおかしい?`canvasElement`のところでエラーが出ます

お手数ですが一度見ていただけませんでしょうか🙇🏻‍♀️
追加でコミットしていただいても大丈夫です

![demo](https://github.com/user-attachments/assets/c163a38d-349e-4e75-9d4c-91c683c2427a)

kenfdev commented:
@namidapoo 
（取り急ぎの回答です）
こちら、私の環境では再現しないのですがおそらく `getByText` の性質で起こる問題かと思います！
`getBy` 系にはリトライ機構が無かったかと思うので、「存在していることが確信できる」ときに使うものになります。
私の環境ではもしかしたらレンダリングだったりタイミングが大丈夫なのかもしれませんが、確信できないタイミングのときは `findBy` 系を使うことになります！こちらはPromiseを返して、自動でリトライもしてくれます。
一度試してみていただけるでしょうか？？

https://testing-library.com/docs/queries/about/#types-of-queries

> わざと落ちるテストを書いた時のエラーメッセージもおかしい?canvasElementのところでエラーが出ます

すみません、こちらもなぜか私の環境で再現しないのですが、エラーを見る限りではgetByTextという認識はできてそうなので、もしかしたらライブラリが報告している行番号が間違っているのかもしれません 🤔 
私も再現できるか試してみようと思いますが、こちらは水曜日以降になってしまうかと思います！ 🙏

namidapoo commented:
私の環境依存かとも思ったんですが、GitHub Actionsが確定で落ちるんですよね🤔
そしてfindByも落ちました..
<details><summary>log</summary>
<p>

```
2025-04-07T00:20:17.0949172Z Current runner version: '2.323.0'
2025-04-07T00:20:17.0974854Z ##[group]Operating System
2025-04-07T00:20:17.0975701Z Ubuntu
2025-04-07T00:20:17.0976205Z 24.04.2
2025-04-07T00:20:17.0976679Z LTS
2025-04-07T00:20:17.0977270Z ##[endgroup]
2025-04-07T00:20:17.0977806Z ##[group]Runner Image
2025-04-07T00:20:17.0978320Z Image: ubuntu-24.04
2025-04-07T00:20:17.0978927Z Version: 20250323.1.0
2025-04-07T00:20:17.0980144Z Included Software: https://github.com/actions/runner-images/blob/ubuntu24/20250323.1/images/ubuntu/Ubuntu2404-Readme.md
2025-04-07T00:20:17.0981585Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu24%2F20250323.1
2025-04-07T00:20:17.0982533Z ##[endgroup]
2025-04-07T00:20:17.0983072Z ##[group]Runner Image Provisioner
2025-04-07T00:20:17.0983676Z 2.0.422.1
2025-04-07T00:20:17.0984192Z ##[endgroup]
2025-04-07T00:20:17.0985246Z ##[group]GITHUB_TOKEN Permissions
2025-04-07T00:20:17.0987270Z Contents: read
2025-04-07T00:20:17.0987848Z Metadata: read
2025-04-07T00:20:17.0988658Z Packages: read
2025-04-07T00:20:17.0989242Z ##[endgroup]
2025-04-07T00:20:17.0992626Z Secret source: Actions
2025-04-07T00:20:17.0993569Z Prepare workflow directory
2025-04-07T00:20:17.1298207Z Prepare all required actions
2025-04-07T00:20:17.1334655Z Getting action download info
2025-04-07T00:20:17.3206458Z ##[group]Download immutable action package 'actions/checkout@v4'
2025-04-07T00:20:17.3207384Z Version: 4.2.2
2025-04-07T00:20:17.3208236Z Digest: sha256:ccb2698953eaebd21c7bf6268a94f9c26518a7e38e27e0b83c1fe1ad049819b1
2025-04-07T00:20:17.3209286Z Source commit SHA: 11bd71901bbe5b1630ceea73d27597364c9af683
2025-04-07T00:20:17.3210248Z ##[endgroup]
2025-04-07T00:20:17.4009629Z Download action repository 'pnpm/action-setup@v4' (SHA:a7487c7e89a18df4991f7f222e4898a00d66ddda)
2025-04-07T00:20:17.6041261Z ##[group]Download immutable action package 'actions/setup-node@v4'
2025-04-07T00:20:17.6041983Z Version: 4.3.0
2025-04-07T00:20:17.6042672Z Digest: sha256:9f9f8171ffc89cb7f6897e615e2b43b46bb55176fd279129eb0088157eace9b2
2025-04-07T00:20:17.6043565Z Source commit SHA: cdca7365b2dadb8aad0a33bc7601856ffabcc48e
2025-04-07T00:20:17.6044192Z ##[endgroup]
2025-04-07T00:20:17.9653872Z Complete job name: test
2025-04-07T00:20:18.0479808Z ##[group]Run actions/checkout@v4
2025-04-07T00:20:18.0481534Z with:
2025-04-07T00:20:18.0482378Z   repository: namidapoo/storybook-hello
2025-04-07T00:20:18.0483702Z   token: ***
2025-04-07T00:20:18.0484476Z   ssh-strict: true
2025-04-07T00:20:18.0485278Z   ssh-user: git
2025-04-07T00:20:18.0486108Z   persist-credentials: true
2025-04-07T00:20:18.0487029Z   clean: true
2025-04-07T00:20:18.0487852Z   sparse-checkout-cone-mode: true
2025-04-07T00:20:18.0488901Z   fetch-depth: 1
2025-04-07T00:20:18.0489711Z   fetch-tags: false
2025-04-07T00:20:18.0490712Z   show-progress: true
2025-04-07T00:20:18.0491555Z   lfs: false
2025-04-07T00:20:18.0492307Z   submodules: false
2025-04-07T00:20:18.0493146Z   set-safe-directory: true
2025-04-07T00:20:18.0494309Z ##[endgroup]
2025-04-07T00:20:18.2209314Z Syncing repository: namidapoo/storybook-hello
2025-04-07T00:20:18.2213222Z ##[group]Getting Git version info
2025-04-07T00:20:18.2214951Z Working directory is '/home/runner/work/storybook-hello/storybook-hello'
2025-04-07T00:20:18.2217105Z [command]/usr/bin/git version
2025-04-07T00:20:18.2241249Z git version 2.49.0
2025-04-07T00:20:18.2269100Z ##[endgroup]
2025-04-07T00:20:18.2282665Z Temporarily overriding HOME='/home/runner/work/_temp/cb8bdad8-0963-4f4f-bfb3-506032828f03' before making global git config changes
2025-04-07T00:20:18.2285371Z Adding repository directory to the temporary git global config as a safe directory
2025-04-07T00:20:18.2289043Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/storybook-hello/storybook-hello
2025-04-07T00:20:18.2323073Z Deleting the contents of '/home/runner/work/storybook-hello/storybook-hello'
2025-04-07T00:20:18.2326797Z ##[group]Initializing the repository
2025-04-07T00:20:18.2330911Z [command]/usr/bin/git init /home/runner/work/storybook-hello/storybook-hello
2025-04-07T00:20:18.2393115Z hint: Using 'master' as the name for the initial branch. This default branch name
2025-04-07T00:20:18.2396271Z hint: is subject to change. To configure the initial branch name to use in all
2025-04-07T00:20:18.2399020Z hint: of your new repositories, which will suppress this warning, call:
2025-04-07T00:20:18.2400732Z hint:
2025-04-07T00:20:18.2401784Z hint: 	git config --global init.defaultBranch <name>
2025-04-07T00:20:18.2402936Z hint:
2025-04-07T00:20:18.2404011Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2025-04-07T00:20:18.2405799Z hint: 'development'. The just-created branch can be renamed via this command:
2025-04-07T00:20:18.2407210Z hint:
2025-04-07T00:20:18.2407946Z hint: 	git branch -m <name>
2025-04-07T00:20:18.2409498Z Initialized empty Git repository in /home/runner/work/storybook-hello/storybook-hello/.git/
2025-04-07T00:20:18.2415898Z [command]/usr/bin/git remote add origin https://github.com/namidapoo/storybook-hello
2025-04-07T00:20:18.2450260Z ##[endgroup]
2025-04-07T00:20:18.2451689Z ##[group]Disabling automatic garbage collection
2025-04-07T00:20:18.2453464Z [command]/usr/bin/git config --local gc.auto 0
2025-04-07T00:20:18.2484036Z ##[endgroup]
2025-04-07T00:20:18.2485325Z ##[group]Setting up auth
2025-04-07T00:20:18.2490088Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2025-04-07T00:20:18.2524111Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2025-04-07T00:20:18.2794038Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2025-04-07T00:20:18.2824171Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2025-04-07T00:20:18.3050845Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2025-04-07T00:20:18.3089589Z ##[endgroup]
2025-04-07T00:20:18.3092584Z ##[group]Fetching the repository
2025-04-07T00:20:18.3102128Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +3cf96ab445c818f71ed8c0f568c4d15ce6afbf24:refs/remotes/pull/4/merge
2025-04-07T00:20:18.5543602Z From https://github.com/namidapoo/storybook-hello
2025-04-07T00:20:18.5546380Z  * [new ref]         3cf96ab445c818f71ed8c0f568c4d15ce6afbf24 -> pull/4/merge
2025-04-07T00:20:18.5568570Z ##[endgroup]
2025-04-07T00:20:18.5570866Z ##[group]Determining the checkout info
2025-04-07T00:20:18.5572637Z ##[endgroup]
2025-04-07T00:20:18.5574421Z [command]/usr/bin/git sparse-checkout disable
2025-04-07T00:20:18.5613610Z [command]/usr/bin/git config --local --unset-all extensions.worktreeConfig
2025-04-07T00:20:18.5640588Z ##[group]Checking out the ref
2025-04-07T00:20:18.5642880Z [command]/usr/bin/git checkout --progress --force refs/remotes/pull/4/merge
2025-04-07T00:20:18.5756579Z Note: switching to 'refs/remotes/pull/4/merge'.
2025-04-07T00:20:18.5757902Z 
2025-04-07T00:20:18.5759097Z You are in 'detached HEAD' state. You can look around, make experimental
2025-04-07T00:20:18.5761887Z changes and commit them, and you can discard any commits you make in this
2025-04-07T00:20:18.5763627Z state without impacting any branches by switching back to a branch.
2025-04-07T00:20:18.5765135Z 
2025-04-07T00:20:18.5765803Z If you want to create a new branch to retain commits you create, you may
2025-04-07T00:20:18.5767483Z do so (now or later) by using -c with the switch command. Example:
2025-04-07T00:20:18.5768406Z 
2025-04-07T00:20:18.5768780Z   git switch -c <new-branch-name>
2025-04-07T00:20:18.5769408Z 
2025-04-07T00:20:18.5769774Z Or undo this operation with:
2025-04-07T00:20:18.5770589Z 
2025-04-07T00:20:18.5770911Z   git switch -
2025-04-07T00:20:18.5771372Z 
2025-04-07T00:20:18.5772400Z Turn off this advice by setting config variable advice.detachedHead to false
2025-04-07T00:20:18.5773479Z 
2025-04-07T00:20:18.5774705Z HEAD is now at 3cf96ab Merge dd9d87c2f098917c3c91d26e84bccd9193e3fa7f into 49afb9ab65cad94e5f8bcdf3c29c7a05c487fcc1
2025-04-07T00:20:18.5778054Z ##[endgroup]
2025-04-07T00:20:18.5800613Z [command]/usr/bin/git log -1 --format=%H
2025-04-07T00:20:18.5821579Z 3cf96ab445c818f71ed8c0f568c4d15ce6afbf24
2025-04-07T00:20:18.6062588Z ##[group]Run pnpm/action-setup@v4
2025-04-07T00:20:18.6063551Z with:
2025-04-07T00:20:18.6064228Z   version: 10.7.1
2025-04-07T00:20:18.6064977Z   dest: ~/setup-pnpm
2025-04-07T00:20:18.6065744Z   run_install: null
2025-04-07T00:20:18.6066537Z   package_json_file: package.json
2025-04-07T00:20:18.6067451Z   standalone: false
2025-04-07T00:20:18.6068189Z ##[endgroup]
2025-04-07T00:20:18.6609096Z ##[group]Running self-installer...
2025-04-07T00:20:19.4889513Z Progress: resolved 1, reused 0, downloaded 0, added 0
2025-04-07T00:20:19.5011899Z Packages: +1
2025-04-07T00:20:19.5014485Z +
2025-04-07T00:20:19.9594461Z Progress: resolved 1, reused 0, downloaded 1, added 1, done
2025-04-07T00:20:20.0307120Z 
2025-04-07T00:20:20.0308357Z dependencies:
2025-04-07T00:20:20.0309129Z + pnpm 10.7.1
2025-04-07T00:20:20.0309457Z 
2025-04-07T00:20:20.0352083Z Done in 1.2s
2025-04-07T00:20:20.0549659Z ##[endgroup]
2025-04-07T00:20:20.0554148Z Installation Completed!
2025-04-07T00:20:20.0702875Z ##[group]Run actions/setup-node@v4
2025-04-07T00:20:20.0703245Z with:
2025-04-07T00:20:20.0703494Z   node-version: 22
2025-04-07T00:20:20.0703758Z   cache: pnpm
2025-04-07T00:20:20.0704009Z   always-auth: false
2025-04-07T00:20:20.0704294Z   check-latest: false
2025-04-07T00:20:20.0704723Z   token: ***
2025-04-07T00:20:20.0705002Z env:
2025-04-07T00:20:20.0705315Z   PNPM_HOME: /home/runner/setup-pnpm/node_modules/.bin
2025-04-07T00:20:20.0705684Z ##[endgroup]
2025-04-07T00:20:20.2411263Z Found in cache @ /opt/hostedtoolcache/node/22.14.0/x64
2025-04-07T00:20:20.2418151Z ##[group]Environment details
2025-04-07T00:20:28.0155038Z node: v22.14.0
2025-04-07T00:20:28.0155437Z npm: 10.9.2
2025-04-07T00:20:28.0155733Z yarn: 1.22.22
2025-04-07T00:20:28.0156656Z ##[endgroup]
2025-04-07T00:20:28.0182165Z [command]/home/runner/setup-pnpm/node_modules/.bin/pnpm store path --silent
2025-04-07T00:20:28.8286322Z /home/runner/setup-pnpm/node_modules/.bin/store/v10
2025-04-07T00:20:28.9380521Z Cache hit for: node-cache-Linux-x64-pnpm-e33ba8d27b459e445ac84f06aa962919944c074825e0c277f24652542c42568f
2025-04-07T00:20:29.9781809Z Received 171278616 of 179667224 (95.3%), 163.2 MBs/sec
2025-04-07T00:20:30.0809510Z Received 179667224 of 179667224 (100.0%), 154.9 MBs/sec
2025-04-07T00:20:30.0811612Z Cache Size: ~171 MB (179667224 B)
2025-04-07T00:20:30.0930974Z [command]/usr/bin/tar -xf /home/runner/work/_temp/3a3c2e39-5331-47e5-8a53-83c1b994374d/cache.tzst -P -C /home/runner/work/storybook-hello/storybook-hello --use-compress-program unzstd
2025-04-07T00:20:31.6725994Z Cache restored successfully
2025-04-07T00:20:31.7113896Z Cache restored from key: node-cache-Linux-x64-pnpm-e33ba8d27b459e445ac84f06aa962919944c074825e0c277f24652542c42568f
2025-04-07T00:20:31.7260476Z ##[group]Run pnpm install
2025-04-07T00:20:31.7260811Z [36;1mpnpm install[0m
2025-04-07T00:20:31.7312537Z shell: /usr/bin/bash -e {0}
2025-04-07T00:20:31.7312795Z env:
2025-04-07T00:20:31.7313038Z   PNPM_HOME: /home/runner/setup-pnpm/node_modules/.bin
2025-04-07T00:20:31.7313344Z ##[endgroup]
2025-04-07T00:20:32.0460597Z Lockfile is up to date, resolution step is skipped
2025-04-07T00:20:32.0996278Z Progress: resolved 1, reused 0, downloaded 0, added 0
2025-04-07T00:20:32.1644782Z Packages: +435
2025-04-07T00:20:32.1645593Z ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
2025-04-07T00:20:32.9962810Z Progress: resolved 435, reused 435, downloaded 0, added 435, done
2025-04-07T00:20:33.2831931Z 
2025-04-07T00:20:33.2832943Z dependencies:
2025-04-07T00:20:33.2833611Z + next 15.2.3
2025-04-07T00:20:33.2834025Z + react 19.0.0
2025-04-07T00:20:33.2834894Z + react-dom 19.0.0
2025-04-07T00:20:33.2835232Z 
2025-04-07T00:20:33.2835461Z devDependencies:
2025-04-07T00:20:33.2835885Z + @chromatic-com/storybook 3.2.6
2025-04-07T00:20:33.2836337Z + @storybook/addon-essentials 8.6.7
2025-04-07T00:20:33.2836728Z + @storybook/addon-onboarding 8.6.7
2025-04-07T00:20:33.2837070Z + @storybook/blocks 8.6.7
2025-04-07T00:20:33.2837492Z + @storybook/experimental-addon-test 8.6.7
2025-04-07T00:20:33.2837951Z + @storybook/experimental-nextjs-vite 8.6.7
2025-04-07T00:20:33.2838572Z + @storybook/react 8.6.7
2025-04-07T00:20:33.2839115Z + @storybook/test 8.6.7
2025-04-07T00:20:33.2839467Z + @tailwindcss/postcss 4.0.15
2025-04-07T00:20:33.2839813Z + @testing-library/jest-dom 6.6.3
2025-04-07T00:20:33.2840363Z + @testing-library/react 16.2.0
2025-04-07T00:20:33.2840622Z + @types/node 20.17.25
2025-04-07T00:20:33.2840848Z + @types/react 19.0.12
2025-04-07T00:20:33.2841080Z + @types/react-dom 19.0.4
2025-04-07T00:20:33.2841331Z + @vitejs/plugin-react 4.3.4
2025-04-07T00:20:33.2841598Z + @vitest/browser 3.0.9
2025-04-07T00:20:33.2842041Z + @vitest/coverage-v8 3.0.9
2025-04-07T00:20:33.2842460Z + jsdom 26.0.0
2025-04-07T00:20:33.2842847Z + playwright 1.51.1
2025-04-07T00:20:33.2843153Z + storybook 8.6.7
2025-04-07T00:20:33.2843367Z + styled-jsx 5.1.6
2025-04-07T00:20:33.2843582Z + tailwindcss 4.0.15
2025-04-07T00:20:33.2843807Z + typescript 5.8.2
2025-04-07T00:20:33.2844064Z + vite 6.2.4
2025-04-07T00:20:33.2844246Z + vitest 3.0.9
2025-04-07T00:20:33.2844365Z 
2025-04-07T00:20:33.2844917Z ╭ Warning ─────────────────────────────────────────────────────────────────────╮
2025-04-07T00:20:33.2845424Z │                                                                              │
2025-04-07T00:20:33.2846420Z │   Ignored build scripts: esbuild, msw, sharp.                                │
2025-04-07T00:20:33.2847424Z │   Run "pnpm approve-builds" to pick which dependencies should be allowed     │
2025-04-07T00:20:33.2848481Z │   to run scripts.                                                            │
2025-04-07T00:20:33.2849412Z │                                                                              │
2025-04-07T00:20:33.2850382Z ╰──────────────────────────────────────────────────────────────────────────────╯
2025-04-07T00:20:33.2850778Z 
2025-04-07T00:20:33.3047787Z Done in 1.5s using pnpm v10.7.1
2025-04-07T00:20:33.3197657Z ##[group]Run pnpm exec playwright install
2025-04-07T00:20:33.3198029Z [36;1mpnpm exec playwright install[0m
2025-04-07T00:20:33.3245360Z shell: /usr/bin/bash -e {0}
2025-04-07T00:20:33.3245613Z env:
2025-04-07T00:20:33.3245859Z   PNPM_HOME: /home/runner/setup-pnpm/node_modules/.bin
2025-04-07T00:20:33.3246161Z ##[endgroup]
2025-04-07T00:20:34.0540952Z Downloading Chromium 134.0.6998.35 (playwright build v1161) from https://cdn.playwright.dev/dbazure/download/playwright/builds/chromium/1161/chromium-linux.zip
2025-04-07T00:20:34.2758989Z |                                                                                |   0% of 164.9 MiB
2025-04-07T00:20:34.3968832Z |■■■■■■■■                                                                        |  10% of 164.9 MiB
2025-04-07T00:20:34.5786278Z |■■■■■■■■■■■■■■■■                                                                |  20% of 164.9 MiB
2025-04-07T00:20:34.7045332Z |■■■■■■■■■■■■■■■■■■■■■■■■                                                        |  30% of 164.9 MiB
2025-04-07T00:20:34.8111023Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                                |  40% of 164.9 MiB
2025-04-07T00:20:34.9178980Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                        |  50% of 164.9 MiB
2025-04-07T00:20:35.1421672Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                |  60% of 164.9 MiB
2025-04-07T00:20:35.2850716Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                        |  70% of 164.9 MiB
2025-04-07T00:20:35.4173080Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                |  80% of 164.9 MiB
2025-04-07T00:20:35.6199561Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■        |  90% of 164.9 MiB
2025-04-07T00:20:35.7969745Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■| 100% of 164.9 MiB
2025-04-07T00:20:38.6506846Z Chromium 134.0.6998.35 (playwright build v1161) downloaded to /home/runner/.cache/ms-playwright/chromium-1161
2025-04-07T00:20:38.6508700Z Downloading Chromium Headless Shell 134.0.6998.35 (playwright build v1161) from https://cdn.playwright.dev/dbazure/download/playwright/builds/chromium/1161/chromium-headless-shell-linux.zip
2025-04-07T00:20:39.7773375Z |                                                                                |   0% of 100.9 MiB
2025-04-07T00:20:41.7463230Z |■■■■■■■■                                                                        |  10% of 100.9 MiB
2025-04-07T00:20:44.5425314Z |■■■■■■■■■■■■■■■■                                                                |  20% of 100.9 MiB
2025-04-07T00:20:47.1914969Z |■■■■■■■■■■■■■■■■■■■■■■■■                                                        |  30% of 100.9 MiB
2025-04-07T00:20:48.4643012Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                                |  40% of 100.9 MiB
2025-04-07T00:20:51.0933331Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                        |  50% of 100.9 MiB
2025-04-07T00:20:52.5328690Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                |  60% of 100.9 MiB
2025-04-07T00:20:54.8870262Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                        |  70% of 100.9 MiB
2025-04-07T00:20:55.8412044Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                |  80% of 100.9 MiB
2025-04-07T00:20:57.6920514Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■        |  90% of 100.9 MiB
2025-04-07T00:20:59.2756390Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■| 100% of 100.9 MiB
2025-04-07T00:21:00.9533260Z Chromium Headless Shell 134.0.6998.35 (playwright build v1161) downloaded to /home/runner/.cache/ms-playwright/chromium_headless_shell-1161
2025-04-07T00:21:00.9536610Z Downloading Firefox 135.0 (playwright build v1475) from https://cdn.playwright.dev/dbazure/download/playwright/builds/firefox/1475/firefox-ubuntu-24.04.zip
2025-04-07T00:21:01.1406678Z |                                                                                |   0% of 90.6 MiB
2025-04-07T00:21:01.2161245Z |■■■■■■■■                                                                        |  10% of 90.6 MiB
2025-04-07T00:21:01.2748011Z |■■■■■■■■■■■■■■■■                                                                |  20% of 90.6 MiB
2025-04-07T00:21:01.3329145Z |■■■■■■■■■■■■■■■■■■■■■■■■                                                        |  30% of 90.6 MiB
2025-04-07T00:21:01.3917433Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                                |  40% of 90.6 MiB
2025-04-07T00:21:01.4575109Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                        |  50% of 90.6 MiB
2025-04-07T00:21:01.5416774Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                |  60% of 90.6 MiB
2025-04-07T00:21:01.5962136Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                        |  70% of 90.6 MiB
2025-04-07T00:21:01.7089203Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                |  80% of 90.6 MiB
2025-04-07T00:21:01.7813417Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■        |  90% of 90.6 MiB
2025-04-07T00:21:01.8357881Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■| 100% of 90.6 MiB
2025-04-07T00:21:03.3404277Z Firefox 135.0 (playwright build v1475) downloaded to /home/runner/.cache/ms-playwright/firefox-1475
2025-04-07T00:21:03.3405993Z Downloading Webkit 18.4 (playwright build v2140) from https://cdn.playwright.dev/dbazure/download/playwright/builds/webkit/2140/webkit-ubuntu-24.04.zip
2025-04-07T00:21:03.4831466Z |                                                                                |   0% of 92 MiB
2025-04-07T00:21:03.6176059Z |■■■■■■■■                                                                        |  10% of 92 MiB
2025-04-07T00:21:03.6847847Z |■■■■■■■■■■■■■■■■                                                                |  20% of 92 MiB
2025-04-07T00:21:03.7476569Z |■■■■■■■■■■■■■■■■■■■■■■■■                                                        |  30% of 92 MiB
2025-04-07T00:21:03.8317951Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                                |  40% of 92 MiB
2025-04-07T00:21:03.8956577Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                        |  50% of 92 MiB
2025-04-07T00:21:03.9562668Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                |  60% of 92 MiB
2025-04-07T00:21:04.0018806Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                        |  70% of 92 MiB
2025-04-07T00:21:04.0675258Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                |  80% of 92 MiB
2025-04-07T00:21:04.1748460Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■        |  90% of 92 MiB
2025-04-07T00:21:04.2428436Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■| 100% of 92 MiB
2025-04-07T00:21:05.7933258Z Webkit 18.4 (playwright build v2140) downloaded to /home/runner/.cache/ms-playwright/webkit-2140
2025-04-07T00:21:05.7934802Z Downloading FFMPEG playwright build v1011 from https://cdn.playwright.dev/dbazure/download/playwright/builds/ffmpeg/1011/ffmpeg-linux.zip
2025-04-07T00:21:05.9284086Z |                                                                                |   0% of 2.3 MiB
2025-04-07T00:21:05.9347404Z |■■■■■■■■                                                                        |  10% of 2.3 MiB
2025-04-07T00:21:05.9358984Z |■■■■■■■■■■■■■■■■                                                                |  20% of 2.3 MiB
2025-04-07T00:21:05.9378104Z |■■■■■■■■■■■■■■■■■■■■■■■■                                                        |  30% of 2.3 MiB
2025-04-07T00:21:05.9395351Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                                |  40% of 2.3 MiB
2025-04-07T00:21:05.9408989Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                        |  50% of 2.3 MiB
2025-04-07T00:21:05.9423283Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                |  60% of 2.3 MiB
2025-04-07T00:21:05.9441328Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                        |  70% of 2.3 MiB
2025-04-07T00:21:05.9458126Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                |  80% of 2.3 MiB
2025-04-07T00:21:05.9470215Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■        |  90% of 2.3 MiB
2025-04-07T00:21:05.9485322Z |■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■| 100% of 2.3 MiB
2025-04-07T00:21:06.0000139Z FFMPEG playwright build v1011 downloaded to /home/runner/.cache/ms-playwright/ffmpeg-1011
2025-04-07T00:21:08.4292844Z Playwright Host validation warning: 
2025-04-07T00:21:08.4293938Z ╔══════════════════════════════════════════════════════╗
2025-04-07T00:21:08.4295129Z ║ Host system is missing dependencies to run browsers. ║
2025-04-07T00:21:08.4295900Z ║ Missing libraries:                                   ║
2025-04-07T00:21:08.4296606Z ║     libgtk-4.so.1                                    ║
2025-04-07T00:21:08.4297292Z ║     libgraphene-1.0.so.0                             ║
2025-04-07T00:21:08.4297982Z ║     libwoff2dec.so.1.0.2                             ║
2025-04-07T00:21:08.4298667Z ║     libvpx.so.9                                      ║
2025-04-07T00:21:08.4299347Z ║     libevent-2.1.so.7                                ║
2025-04-07T00:21:08.4300229Z ║     libopus.so.0                                     ║
2025-04-07T00:21:08.4301203Z ║     libgstallocators-1.0.so.0                        ║
2025-04-07T00:21:08.4301904Z ║     libgstapp-1.0.so.0                               ║
2025-04-07T00:21:08.4302606Z ║     libgstpbutils-1.0.so.0                           ║
2025-04-07T00:21:08.4303313Z ║     libgstaudio-1.0.so.0                             ║
2025-04-07T00:21:08.4304005Z ║     libgstgl-1.0.so.0                                ║
2025-04-07T00:21:08.4304682Z ║     libgsttag-1.0.so.0                               ║
2025-04-07T00:21:08.4305383Z ║     libgstvideo-1.0.so.0                             ║
2025-04-07T00:21:08.4306090Z ║     libgstcodecparsers-1.0.so.0                      ║
2025-04-07T00:21:08.4306847Z ║     libgstfft-1.0.so.0                               ║
2025-04-07T00:21:08.4307541Z ║     libflite.so.1                                    ║
2025-04-07T00:21:08.4308238Z ║     libflite_usenglish.so.1                          ║
2025-04-07T00:21:08.4308942Z ║     libflite_cmu_grapheme_lang.so.1                  ║
2025-04-07T00:21:08.4309658Z ║     libflite_cmu_grapheme_lex.so.1                   ║
2025-04-07T00:21:08.4310525Z ║     libflite_cmu_indic_lang.so.1                     ║
2025-04-07T00:21:08.4311248Z ║     libflite_cmu_indic_lex.so.1                      ║
2025-04-07T00:21:08.4311961Z ║     libflite_cmulex.so.1                             ║
2025-04-07T00:21:08.4312655Z ║     libflite_cmu_time_awb.so.1                       ║
2025-04-07T00:21:08.4313347Z ║     libflite_cmu_us_awb.so.1                         ║
2025-04-07T00:21:08.4314034Z ║     libflite_cmu_us_kal16.so.1                       ║
2025-04-07T00:21:08.4314722Z ║     libflite_cmu_us_kal.so.1                         ║
2025-04-07T00:21:08.4315425Z ║     libflite_cmu_us_rms.so.1                         ║
2025-04-07T00:21:08.4316119Z ║     libflite_cmu_us_slt.so.1                         ║
2025-04-07T00:21:08.4316807Z ║     libavif.so.16                                    ║
2025-04-07T00:21:08.4317502Z ║     libharfbuzz-icu.so.0                             ║
2025-04-07T00:21:08.4318220Z ║     libsecret-1.so.0                                 ║
2025-04-07T00:21:08.4318915Z ║     libhyphen.so.0                                   ║
2025-04-07T00:21:08.4319607Z ║     libmanette-0.2.so.0                              ║
2025-04-07T00:21:08.4320656Z ║     libGLESv2.so.2                                   ║
2025-04-07T00:21:08.4321381Z ║     libx264.so                                       ║
2025-04-07T00:21:08.4322104Z ╚══════════════════════════════════════════════════════╝
2025-04-07T00:21:08.4323637Z     at validateDependenciesLinux (/home/runner/work/storybook-hello/storybook-hello/node_modules/.pnpm/playwright-core@1.51.1/node_modules/playwright-core/lib/server/registry/dependencies.js:216:9)
2025-04-07T00:21:08.4326099Z     at async Registry._validateHostRequirements (/home/runner/work/storybook-hello/storybook-hello/node_modules/.pnpm/playwright-core@1.51.1/node_modules/playwright-core/lib/server/registry/index.js:859:52)
2025-04-07T00:21:08.4328741Z     at async Registry._validateHostRequirementsForExecutableIfNeeded (/home/runner/work/storybook-hello/storybook-hello/node_modules/.pnpm/playwright-core@1.51.1/node_modules/playwright-core/lib/server/registry/index.js:957:7)
2025-04-07T00:21:08.4331758Z     at async Registry.validateHostRequirementsForExecutablesIfNeeded (/home/runner/work/storybook-hello/storybook-hello/node_modules/.pnpm/playwright-core@1.51.1/node_modules/playwright-core/lib/server/registry/index.js:946:43)
2025-04-07T00:21:08.4334210Z     at async t.<anonymous> (/home/runner/work/storybook-hello/storybook-hello/node_modules/.pnpm/playwright-core@1.51.1/node_modules/playwright-core/lib/cli/program.js:122:7)
2025-04-07T00:21:08.4465964Z ##[group]Run pnpm test --passWithNoTests
2025-04-07T00:21:08.4466320Z [36;1mpnpm test --passWithNoTests[0m
2025-04-07T00:21:08.4513323Z shell: /usr/bin/bash -e {0}
2025-04-07T00:21:08.4513549Z env:
2025-04-07T00:21:08.4513790Z   PNPM_HOME: /home/runner/setup-pnpm/node_modules/.bin
2025-04-07T00:21:08.4514248Z ##[endgroup]
2025-04-07T00:21:08.7334125Z 
2025-04-07T00:21:08.7335153Z > storybook-hello@0.1.0 test /home/runner/work/storybook-hello/storybook-hello
2025-04-07T00:21:08.7335957Z > vitest --passWithNoTests
2025-04-07T00:21:08.7336246Z 
2025-04-07T00:21:11.7680904Z info Using tsconfig paths for react-docgen
2025-04-07T00:21:12.3363731Z [1m[7m[33m Vitest [39m[27m[22m [33mNo browser "instances" were defined for the "0" project. Running tests in "chromium" browser. The "browser.name" field is deprecated since Vitest 3. Read more: https://vitest.dev/guide/browser/config#browser-instances[39m
2025-04-07T00:21:12.3383019Z 
2025-04-07T00:21:12.3384251Z [1m[7m[36m RUN [39m[27m[22m [36mv3.0.9 [39m[90m/home/runner/work/storybook-hello/storybook-hello[39m
2025-04-07T00:21:12.3384886Z 
2025-04-07T00:21:12.4423375Z    Loading config from /home/runner/work/storybook-hello/storybook-hello/next.config.ts
2025-04-07T00:21:12.4714464Z    Loading config from /home/runner/work/storybook-hello/storybook-hello/next.config.ts
2025-04-07T00:21:12.5811940Z Failed to resolve dependency: [36m@mdx-js/react[39m, present in client 'optimizeDeps.include'
2025-04-07T00:21:12.5821979Z Failed to resolve dependency: [36m@mdx-js/react[39m, present in client 'optimizeDeps.include'
2025-04-07T00:21:13.5175096Z  [32m✓[39m [30m[46m 1 [49m[39m app/utils/sum.test.ts [2m([22m[2m1 test[22m[2m)[22m[90m 7[2mms[22m[39m
2025-04-07T00:21:13.8609497Z  [32m✓[39m [30m[46m 1 [49m[39m app/page.test.tsx [2m([22m[2m1 test[22m[2m)[22m[90m 101[2mms[22m[39m
2025-04-07T00:21:23.2246944Z  [32m✓[39m [30m[45m 0 (chromium) [49m[39m stories/Header.stories.ts [2m([22m[2m2 tests[22m[2m)[22m[90m 244[2mms[22m[39m
2025-04-07T00:21:23.3323445Z  [32m✓[39m [30m[45m 0 (chromium) [49m[39m stories/Page.stories.ts [2m([22m[2m4 tests[22m[2m)[22m[33m 314[2mms[22m[39m
2025-04-07T00:21:23.4044861Z  [32m✓[39m [30m[45m 0 (chromium) [49m[39m stories/ParamsClientComponent.stories.tsx [2m([22m[2m5 tests[22m[2m)[22m[33m 483[2mms[22m[39m
2025-04-07T00:21:23.4046473Z    [33m[2m✓[22m[39m Default [33m408[2mms[22m[39m
2025-04-07T00:21:24.0072258Z  [32m✓[39m [30m[45m 0 (chromium) [49m[39m stories/ParamsServerComponent.stories.tsx [2m([22m[2m5 tests[22m[2m)[22m[90m 152[2mms[22m[39m
2025-04-07T00:21:24.0379244Z  [32m✓[39m [30m[45m 0 (chromium) [49m[39m stories/Button.stories.ts [2m([22m[2m4 tests[22m[2m)[22m[90m 76[2mms[22m[39m
2025-04-07T00:21:24.3832686Z [2m12:21:24 AM[22m [36m[1m[vite][22m[39m [90m[2m(client)[22m[39m [32m✨ new dependencies optimized: [33msb-original/default-loader, sb-original/image-context[32m[39m
2025-04-07T00:21:24.3833919Z [33m
2025-04-07T00:21:24.3835137Z [1m[vitest][22m Vite unexpectedly reloaded a test. This may cause tests to fail, lead to flaky behaviour or duplicated test runs.
2025-04-07T00:21:24.3836901Z [39m[33mFor a stable experience, please add mentioned dependencies to your config's [1m`optimizeDeps.include`[22m field manually.
2025-04-07T00:21:24.3837704Z 
2025-04-07T00:21:24.3837872Z [39m
2025-04-07T00:21:24.3838874Z [2m12:21:24 AM[22m [36m[1m[vite][22m[39m [90m[2m(client)[22m[39m [32m✨ optimized dependencies changed. reloading[39m
2025-04-07T00:21:24.4927633Z [90mstderr[2m | stories/example.stories.tsx[2m > [22m[2mDefault
2025-04-07T00:21:24.4929233Z [22m[39mAn update to %s inside a test was not wrapped in act(...).
2025-04-07T00:21:24.4930041Z 
2025-04-07T00:21:24.4930785Z When testing, code that causes React state updates should be wrapped into act(...):
2025-04-07T00:21:24.4931393Z 
2025-04-07T00:21:24.4931537Z act(() => {
2025-04-07T00:21:24.4932323Z   /* fire events that update state */
2025-04-07T00:21:24.4932741Z });
2025-04-07T00:21:24.4932985Z /* assert on the output */
2025-04-07T00:21:24.4933157Z 
2025-04-07T00:21:24.4933874Z This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act Root
2025-04-07T00:21:24.4969158Z [90mstderr[2m | stories/example.stories.tsx[2m > [22m[2mDefault
2025-04-07T00:21:24.4971204Z [22m[39mInvalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
2025-04-07T00:21:24.4972642Z 1. You might have mismatching versions of React and the renderer (such as React DOM)
2025-04-07T00:21:24.4973486Z 2. You might be breaking the Rules of Hooks
2025-04-07T00:21:24.4974222Z 3. You might have more than one copy of React in the same app
2025-04-07T00:21:24.4975312Z See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.
2025-04-07T00:21:24.4999256Z [90mstderr[2m | stories/example.stories.tsx[2m > [22m[2mDefault
2025-04-07T00:21:24.5001155Z [22m[39mInvalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
2025-04-07T00:21:24.5002573Z 1. You might have mismatching versions of React and the renderer (such as React DOM)
2025-04-07T00:21:24.5003425Z 2. You might be breaking the Rules of Hooks
2025-04-07T00:21:24.5004144Z 3. You might have more than one copy of React in the same app
2025-04-07T00:21:24.5005221Z See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.
2025-04-07T00:21:24.5024061Z [90mstderr[2m | stories/example.stories.tsx[2m > [22m[2mDefault
2025-04-07T00:21:24.5026111Z [22m[39m[TypeError: Cannot read properties of null (reading 'useMemo')]
2025-04-07T00:21:24.5031148Z [90mstderr[2m | stories/example.stories.tsx[2m > [22m[2mDefault
2025-04-07T00:21:24.5033502Z [22m[39m%s
2025-04-07T00:21:24.5033749Z 
2025-04-07T00:21:24.5033892Z %s
2025-04-07T00:21:24.5034928Z  An error occurred in the <HeadManagerProvider> component. Consider adding an error boundary to your tree to customize error handling behavior.
2025-04-07T00:21:24.5036411Z Visit https://react.dev/link/error-boundaries to learn more about error boundaries.
2025-04-07T00:21:24.5092768Z  [31m❯[39m [30m[45m 0 (chromium) [49m[39m stories/example.stories.tsx [2m([22m[2m1 test[22m[2m | [22m[31m1 failed[39m[2m)[22m[90m 96[2mms[22m[39m
2025-04-07T00:21:24.5094132Z [31m   [31m×[31m Default[90m 96[2mms[22m[31m[39m
2025-04-07T00:21:24.5094687Z [31m     → 
2025-04-07T00:21:24.5096030Z [34mClick to debug the error directly in Storybook: http://localhost:6006/?path=/story/pages-home--default&addonPanel=storybook/test/panel[31m
2025-04-07T00:21:24.5096924Z 
2025-04-07T00:21:24.5097120Z expect(received).toBeInTheDocument()
2025-04-07T00:21:24.5097317Z 
2025-04-07T00:21:24.5097541Z received value must be an HTMLElement or an SVGElement.
2025-04-07T00:21:24.5097887Z [39m
2025-04-07T00:21:24.5143311Z 
2025-04-07T00:21:24.5143894Z [31m⎯⎯⎯⎯⎯⎯⎯[1m[7m Failed Tests 1 [27m[22m⎯⎯⎯⎯⎯⎯⎯[39m
2025-04-07T00:21:24.5144256Z 
2025-04-07T00:21:24.5146550Z [31m[1m[7m FAIL [27m[22m[39m [30m[45m 0 (chromium) [49m[39m stories/example.stories.tsx[2m > [22mDefault
2025-04-07T00:21:24.5176754Z [31m[1mError[22m: 
2025-04-07T00:21:24.5178279Z [34mClick to debug the error directly in Storybook: http://localhost:6006/?path=/story/pages-home--default&addonPanel=storybook/test/panel[31m
2025-04-07T00:21:24.5179172Z 
2025-04-07T00:21:24.5179471Z expect(received).toBeInTheDocument()
2025-04-07T00:21:24.5179780Z 
2025-04-07T00:21:24.5180419Z received value must be an HTMLElement or an SVGElement.
2025-04-07T00:21:24.5181048Z [39m
2025-04-07T00:21:24.5181798Z [36m [2m❯[22m play stories/example.stories.tsx:[2m18:53[22m[39m
2025-04-07T00:21:24.5206251Z     [90m 16| [39m
2025-04-07T00:21:24.5207351Z     [90m 17| [39m[35mexport[39m [35mconst[39m [33mDefault[39m[33m:[39m [33mStory[39m [33m=[39m {
2025-04-07T00:21:24.5208383Z     [90m 18| [39m play[33m:[39m [35masync[39m ({ canvasElement }) [33m=>[39m {
2025-04-07T00:21:24.5209186Z     [90m   | [39m                                                    [31m^[39m
2025-04-07T00:21:24.5210218Z     [90m 19| [39m  [35mconst[39m canvas [33m=[39m [34mwithin[39m(canvasElement)[33m;[39m
2025-04-07T00:21:24.5211668Z     [90m 20| [39m  [35mawait[39m [34mexpect[39m(canvas[33m.[39m[34mfindByText[39m([32m"app/page.tsx"[39m))[33m.[39m[34mtoBeInTheDocument[39m()[33m;[39m
2025-04-07T00:21:24.5213448Z [90m [2m❯[22m node_modules/.vite/deps/@storybook_experimental-addon-test_internal_test-utils.js?v=3a74c102:[2m57:110[22m[39m
2025-04-07T00:21:24.5214217Z 
2025-04-07T00:21:24.5214607Z [31m[2m⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯[22m[39m
2025-04-07T00:21:24.5215037Z 
2025-04-07T00:21:24.5215050Z 
2025-04-07T00:21:24.5215758Z [2m Test Files [22m [1m[31m1 failed[39m[22m[2m | [22m[1m[32m7 passed[39m[22m[90m (8)[39m
2025-04-07T00:21:24.5216918Z [2m      Tests [22m [1m[31m1 failed[39m[22m[2m | [22m[1m[32m22 passed[39m[22m[90m (23)[39m
2025-04-07T00:21:24.5217681Z [2m   Start at [22m 00:21:12
2025-04-07T00:21:24.5218851Z [2m   Duration [22m 12.18s[2m (transform 225ms, setup 5.34s, collect 1.73s, tests 1.47s, environment 1.19s, prepare 2.77s)[22m
2025-04-07T00:21:24.5219521Z 
2025-04-07T00:21:24.5230154Z 
2025-04-07T00:21:24.5261690Z ##[error]Error: 
Click to debug the error directly in Storybook: http://localhost:6006/?path=/story/pages-home--default&addonPanel=storybook/test/panel

expect(received).toBeInTheDocument()

received value must be an HTMLElement or an SVGElement.

 ❯ play stories/example.stories.tsx:18:53
 ❯ node_modules/.vite/deps/@storybook_experimental-addon-test_internal_test-utils.js?v=3a74c102:57:110


2025-04-07T00:21:25.0641566Z  ELIFECYCLE  Test failed. See above for more details.
2025-04-07T00:21:25.0999766Z ##[error]Process completed with exit code 1.
2025-04-07T00:21:25.1064571Z Post job cleanup.
2025-04-07T00:21:25.1599995Z Pruning is unnecessary.
2025-04-07T00:21:25.1706144Z Post job cleanup.
2025-04-07T00:21:25.2650475Z [command]/usr/bin/git version
2025-04-07T00:21:25.2686405Z git version 2.49.0
2025-04-07T00:21:25.2736920Z Temporarily overriding HOME='/home/runner/work/_temp/24285ac0-01a9-4f37-a2e1-8d6fcc096dd5' before making global git config changes
2025-04-07T00:21:25.2738411Z Adding repository directory to the temporary git global config as a safe directory
2025-04-07T00:21:25.2743365Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/storybook-hello/storybook-hello
2025-04-07T00:21:25.2777952Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2025-04-07T00:21:25.2811104Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2025-04-07T00:21:25.3044578Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2025-04-07T00:21:25.3066907Z http.https://github.com/.extraheader
2025-04-07T00:21:25.3085192Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2025-04-07T00:21:25.3120443Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2025-04-07T00:21:25.3473353Z Cleaning up orphan processes
```

</p>
</details>

kenfdev commented on stories/example.stories.tsx:
```diff
@@ -0,0 +1,22 @@
+export const Default: Story = {
+	play: async ({ canvasElement }) => {
+		const canvas = within(canvasElement);
+		await expect(canvas.findByText("app/page.tsx")).toBeInTheDocument();
```
```suggestion
                const text = await canvas.findByText("app/page.tsx");
		await expect(text).toBeInTheDocument();
```

findByTextは `await` してあげないといけないかと思います！

namidapoo commented:
@kenfdev やはりだめそうです...! https://github.com/namidapoo/storybook-hello/pull/4/commits/8943b9c5d10da2506aded6304e1c3c0f2b2116e8
すみませんがお時間あるとき確認お願いします

kenfdev commented:
@namidapoo 
エラーは変わったみたいですね！（getByのときと似たようなものですが...）

> Unable to find an element with the text: app/page.tsx. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.

これは謎ですね。。。今ローカルで試したら必ず私の環境では通っているのですが、鈴木さんの環境もローカルだと通りますか？？（それともまだFlakyでしょうか？？）

私はLinuxで開発しているのでGHAと同じ環境での実行になっているはずなのですが...これは水曜日に時間をとって調査しないといけない気がしてきました！
調査の際このbrancにpushなどたくさんしてしまってもいいですか？？

namidapoo commented:
@kenfdev 
遅くなりました！

おっしゃる通りFlakyというか安定しないというか
以下の流れ踏むと初回は必ず?こけます
```sh
rm -rf node_modules
pnpm i
pnpm test
```
でもそのまま再実行したり、失敗したテストファイルで`cmd + s`で保存（によって再実行が走る）したりすると成功します🤔
ローカルで出るエラーメッセージはGHAのものと同じです！

> 私はLinuxで開発しているのでGHAと同じ環境での実行になっているはずなのですが...これは水曜日に時間をとって調査しないといけない気がしてきました！
調査の際このbrancにpushなどたくさんしてしまってもいいですか？？

毎度お手数おかけします🙇🏻‍♀️ なんでもお好きにやっちゃってください！

kenfdev commented:
@namidapoo 

なるほどです！

```
rm -rf node_modules
pnpm i
pnpm test
```

こちらで私も再現しました、ありがとうございます！

↓そしてこちらで修正しました！
https://github.com/namidapoo/storybook-hello/pull/4/commits/13241169bfa79d77f2d723964290678d96d41ea5

よくよくログを見ると↓というのが出ているのに気づきました

```
[vite] (client) ✨ new dependencies optimized: sb-original/default-loader, sb-original/image-context
...
[vitest] Vite unexpectedly reloaded a test. This may cause tests to fail, lead to flaky behaviour or duplicated test runs.
For a stable experience, please add mentioned dependencies to your config's `optimizeDeps.include` field manually.
```

どうもvitestがdependencyを最適化している処理に、 `next/image` の依存関係が含まれていないみたいです。そのせいで、 `next/image` がうまく読み込まれずに `Home` のレンダリングが初回テストのときに間に合わない（？）ようです。

そしてこの最適化処理は1回目しか行わないので、鈴木さんの手順でのみ再現するみたいですね。試しに `next/image` のコンポーネントを取り除くと、今回の修正をしなくてもテストは通るようになります。

恒久対応としては、 `optimizeDeps` に対象のライブラリを追加して、必ず最適化処理の中に含めることです。

なんとなくですが、こちらはvitest側の不具合な気がするのでIssueは作っておきました！

https://github.com/storybookjs/storybook/issues/31085

また、vitestが報告する行数が合っていない問題も、再現できました！（なぜ私の環境で最初再現しなかったのか謎ですが...）

おそらくなのですがこちらは `@storybook/experimental-nextjs-vite` が影響している気がします。この `framework` を使わない場合は行番号が合っている様です。（別途でvite-reactな環境で試してみました）
こちらはすぐに直すのは難しそうなので、以下にIssueを作っておきました 🙏 

https://github.com/storybookjs/storybook/issues/31081

いったんはエラー文言から場所を特定しないといけなさそうです 💦

namidapoo commented:
@kenfdev 早速調査いただきありがとうございます！
> どうもvitestがdependencyを最適化している処理に、 next/image の依存関係が含まれていないみたいです。そのせいで、 next/image がうまく読み込まれずに Home のレンダリングが初回テストのときに間に合わない（？）ようです。

そういうことだったんですね...!
https://github.com/namidapoo/storybook-hello/pull/4/commits/13241169bfa79d77f2d723964290678d96d41ea5 をpullしたら私のローカル環境でも失敗することなくなりました🙏🏻

issueもありがとうございます!!
行番号が異なるのもデバッグがしにくくて結構しんどかったのでissueあげていただけてとても助かります🙇🏻‍♀️

このあとはissue見守りたいと思います!
迅速に対応いただき本当にありがとうございました🙇🏻‍♀️
